src/main/java/com/example/gymApp/dto/trainee/TraineeMapper.java

```diff
@@ -0,0 +1,92 @@
+
+
+package com.example.gymApp.dto.trainee;
+
+import com.example.gymApp.model.Trainee;
+import java.util.List;
+import org.mapstruct.IterableMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Named;
+
+@Mapper(componentModel = "spring")
+public interface TraineeMapper {
+
+  @Mapping(source = "firstName", target = "user.firstName")
+  @Mapping(source = "lastName", target = "user.lastName")
+  @Mapping(source = "dateOfBirth", target = "dateOfBirth")
+  @Mapping(source = "address", target = "address")
+  Trainee toTrainee(TraineeDto dto);
+
+  @Mapping(source = "user.firstName", target = "firstName")
+  @Mapping(source = "user.lastName", target = "lastName")
+  @Mapping(source = "dateOfBirth", target = "dateOfBirth")
+  @Mapping(source = "address", target = "address")
+  TraineeDto toTraineeDto(Trainee trainee);
+
+
+  @Mapping(source = "user.firstName", target = "firstName")
+  @Mapping(source = "user.lastName", target = "lastName")
+  @Mapping(source = "user.username", target = "username")
+  @Mapping(source = "dateOfBirth", target = "dateOfBirth")
+  @Mapping(source = "address", target = "address")
+  // !!!! In MapStruct, for boolean fields, if the field is called isActive,
+  // MapStruct automatically removes the is prefix and expects the mapper to use
+  // the name active, even if the class name is isActive. This is a built-in MapStruct
+  // behavior that cannot be changed directly without additional configuration.
+  @Mapping(source = "user.active", target = "active")
+  TraineeDto toTraineeDto6Fields(Trainee trainee);
+
+
+  @Named("toTraineeDtoWith3Fields")
+  @Mapping(target = "isActive", ignore = true)
+
+  default TraineeDto toTraineeDtoWith3Fields(Trainee trainee) {
+    TraineeDto dto = new TraineeDto();
+    dto.setFirstName(trainee.getUser().getFirstName());
+    dto.setLastName(trainee.getUser().getLastName());
+    dto.setUsername(trainee.getUser().getUsername());
+
+    return dto;
+  }
```

Comment:
strange implementation, where is toTraineeDtoWith3Fields used?

Reply:
8. Get Trainer Profile (GET method)
a. Request
I. Username (required)
b. Response
I. First Name
II. Last Name
III. Specialization (Training type reference)
IV. Is Active
V. Trainees List
   1. Trainee Username
   2. Trainee First Name
   3. Trainee Last Name
   */